arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -647,36 +644,35 @@ public infix fun <A> Resource<A>.releaseCase(release: suspend (A, ExitCase) -> U
  * suspend fun fileToString(file: File): String = file.toString()
  *
  * suspend fun main(): Unit {
- *   //sampleStart
  *   val res: List<String> = listOf(
  *     "data.json",
  *     "user.json",
  *     "resource.json"
- *   ).traverseResource { uri ->
+ *   ).traverse { uri ->
  *     resource {
  *      openFile(uri)
  *     } release { file ->
  *       closeFile(file)
  *     }
  *   }.use { files ->
  *     files.map { fileToString(it) }
- *   }
- *   //sampleEnd
- *   res.forEach(::println)
+ *   }.forEach(::println)
  * }
  * ```
  * <!--- KNIT example-resource-10.kt -->
  */
-public inline fun <A, B> Iterable<A>.traverseResource(crossinline f: (A) -> Resource<B>): Resource<List<B>> =
+@OptIn(ExperimentalTypeInference::class)
+@OverloadResolutionByLambdaReturnType
+public inline fun <A, B> Iterable<A>.traverse(crossinline transform: (A) -> Resource<B>): Resource<List<B>> =
```

Comment:
```suggestion
public inline fun <A, B> Iterable<A>.traverse(crossinline f: (A) -> Resource<B>): Resource<List<B>> =
```
to keep the parameter name the same across the other `traverse` functions

Reply:
We should've not named the new APIs with `f` ><

Very annoying that this is a API breaking change in Kotlin, and there are not strategies for refactoring/deprecating this. Arrow should try to stick to Kotlin Std naming for parameters as well, but given that there is no way to migrate this not sure how we could improve it for Arrow 2.0.

Reply:
I had to make a manual commit @i-walker since the change from `transform` to `f` was not represented for the method body in the suggestion.